addons/point_of_sale/static/src/app/screens/receipt_screen/receipt_screen.js

```diff
@@ -38,21 +38,20 @@ export class ReceiptScreen extends Component {
     _addNewOrder() {
         this.pos.add_new_order();
     }
-    actionSendReceipt() {
-        if (this.state.mode === "email" && this.isValidEmail(this.state.input)) {
-            this.sendReceipt.call({ action: "action_send_receipt", name: "Email" });
+    actionSendReceiptOnEmail() {
+        if (this.isValidEmailAddress(this.state.email)) {
+            this.sendReceipt.call({ action: "action_send_receipt", destination: this.state.email });
         } else {
             this.notification.add(_t("Please enter a valid email address"), {
                 type: "danger",
             });
         }
     }
-    changeMode(mode) {
-        this.state.mode = mode;
-        this.state.input = this.currentOrder.partner_id?.email || this.state.input || "";
+    get isValidEmail() {
+        return this.isValidEmailAddress(this.state.email);
     }
-    get isValidInput() {
-        return this.isValidEmail(this.state.input);
+    get isValidPhone() {
+        return this.isValidPhoneNumber(this.state.phone);
```

Comment:
why do you need these getters ? You can just call `isValidEmailAddress` and `isValidPhoneNumber` directly